Add column "Views" next to "Total"

maitandat1507 opened the pull request.

maitandat1507 committed:
Add column "Views" next to "Total"
Show data from tinhte_xentag_view_count

maitandat1507 committed:
Add column "Views" next to "Total"
To show data from "tinhte_xentag_view_count"
Add function Ordered By: "Views"

maitandat1507 committed:
Add column "Views" next to "Total"
#fixed: show/hide "Views" in dropdownList "Ordered By" when option Tinhte_XenTag_logView turn on/off

daohoangson commented:
Cái XenForo_Model_Tag không được thay cả hàm như vậy nha. Nếu làm thế thì add-on khác extend cùng hàm đó sẽ không chạy được. Giải pháp hôm trước có giải thích với em rồi đó:
1. Override getTagList nhưng chỉ keep track lại order được yêu cầu thôi, sau đó gọi parent xử lý tiếp
2. Override limitQueryResults, kiểm tra xem có phải đang gọi getTagList với order=tinhte_xentag_view_count không thì dùng str_replace để cập nhật lại phần "ORDER BY" của query

maitandat1507 commented:
Dạ okie anh để em xem lại ạ.👌

Vào 15:42 Ngày 07 tháng 01 năm 2016, Đào Hoàng Sơn <notifications@github.com

> đã viết:
> 
> Cái XenForo_Model_Tag không được thay cả hàm như vậy nha. Nếu làm thế thì
> add-on khác extend cùng hàm đó sẽ không chạy được. Giải pháp hôm trước có
> giải thích với em rồi đó:
> 1. Override getTagList nhưng chỉ keep track lại order được yêu cầu
>    thôi, sau đó gọi parent xử lý tiếp
> 2. Override limitQueryResults, kiểm tra xem có phải đang gọi
>    getTagList với order=tinhte_xentag_view_count không thì dùng str_replace để
>    cập nhật lại phần "ORDER BY" của query
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/xfrocks/Tinhte_XenTag/pull/2#issuecomment-169598572.

maitandat1507 committed:
Add column "Views" next to "Total"
#edited

maitandat1507 committed:
Add column "Views" next to "Total"
#edited 2

maitandat1507 committed:
Add column "Views" next to "Total"
#edited 3

daohoangson commented:
Thấy okie rồi, để last test lên rồi merge nha :+1:

maitandat1507 committed:
Merge branch 'master' of https://github.com/xfrocks/Tinhte_XenTag

daohoangson commented on library/Tinhte/XenTag/addon-Tinhte_XenTag.xml:
```diff
@@ -87,6 +87,31 @@
+$0]]></replace>
+    </modification>
+    <modification template="tag_list" modification_key="tinhte_xentag_add_option_view" description="Add option &quot;Views&quot; in &quot;Ordered By&quot;" execution_order="10" enabled="1" action="str_replace">
+      <find><![CDATA[<xen:select name="order" value="{xen:if $order, $order, tag}">
```
Cái `find` này xem dùng regular expression được không nhỉ? Sau này XenForo thêm lựa chọn vào cái này thì sợ là modification của mình hết chạy được.

daohoangson commented on library/Tinhte/XenTag/addon-Tinhte_XenTag.xml:
```diff
@@ -575,6 +607,7 @@ step=7</edit_format_params>
     <phrase title="tinhte_xentag_new_tagged_resource_x" version_id="115" version_string="1.9.1-dev"><![CDATA[{title} - New resource with watched tag]]></phrase>
     <phrase title="tinhte_xentag_new_tagged_thread_x" version_id="115" version_string="1.9.1-dev"><![CDATA[{title} - New thread with watched tag]]></phrase>
     <phrase title="tinhte_xentag_new_tags" version_id="3010500" version_string="3.1.5"><![CDATA[New Tags]]></phrase>
+    <phrase title="tinhte_xentag_option_views" version_id="3020000" version_string="3.2.0"><![CDATA[Views]]></phrase>
```
Cân nhắc bỏ cái phrase này, xài phrase của hệ thống luôn cho tiện (hệ thống có sẵn cái "views")

maitandat1507 committed:
# Fixed
- Using regular expression to add column Views
- Change phrase "tinhte_xentag..._view" to "views"
